datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -422,29 +422,6 @@ impl Stream for GroupedHashAggregateStream {
     ) -> Poll<Option<Self::Item>> {
         let elapsed_compute = self.baseline_metrics.elapsed_compute().clone();
 
-        // common function for signalling end of processing of the input stream
-        fn set_input_done_and_produce_output(
```

Comment:
This function only modifies fields on `GroupedHashAggregateStream` so we can reduce a level of indirection and follow the pattern in other functionality in this module by moving it to the `impl GroupedHashAggregateStream { ... } `